app/Http/Controllers/AdminController.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\AddAdminRequest;
+use App\Http\Requests\UpdateAdminRequest;
+use App\Http\Resources\PermissionResource;
+use App\Models\Admin;
+use App\Services\AdminService;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+
+class AdminController extends Controller
+{
+    protected $adminService;
+
+    public function __construct(AdminService $adminService)
+    {
+        $this->adminService = $adminService;
+    }
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(AddAdminRequest $request)
+    {
+        $this->authorize('create', Admin::class);
+        $this->adminService->store($request);
```

Comment:
The data should be sent to service's functions as array. Please edit this in all functions
`$request->all()`